src/python/pants/backend/explorer/graphql/query/targets.py

```diff
@@ -136,10 +136,10 @@ def filter(query: TargetsQuery | None, targets: Iterable[TargetData]) -> Iterato
         for data in targets:
             if query.target_type and data.target.alias != query.target_type:
                 continue
+            if query.limit is not None and count >= query.limit:
+                return
```

Comment:
The tests exposed a bug in how the `limit` was applied.

Reply:
Ditto placing the limit before the filtering line.